### PR TITLE
lua: Add sandbox for lua detection scripts

### PR DESCRIPTION
**Pull request by J0eJ0h**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/4777

Describe Changes:

Add sandboxing support for lua detection scripts to avoid memory\cpu overruns and os interaction

Fixes CI issue in https://github.com/OISF/suricata/pull/10272

Includes pending PR https://github.com/OISF/suricata/pull/10263

### CODE REVIEW

**Comment by victorjulien** [comment]
Have you looked into why Centos7 is failing?

**Comment by jasonish** [comment]
Looks like we're getting there.  If I have this right:
- This implements the sandbox for Lua rules
- The API that rules use does not change
- All the Lua S-V tests we have still pass

Another important piece would be a discussion in the documentation about what the sandbox implements, what users need to be aware of, probably in this existing chapter here: https://docs.suricata.io/en/latest/lua/lua-usage.html#lua-detection

Could be bare minimal to start, but with more discussion added later.

Then I'd probably want S-V tests to prove that what is documented actually is.

**Comment by J0eJ0h** [comment]
Updating the docs and tests makes sense.  It's still missing the config to allow bypassing the sandbox for rules that need additional access (e.g. file system).

I think the CentOS issue was that --disable-lua for the configure script was not being honored for distcheck.  I'll try and at least make configure fail if lua is on, but not 5.4.

**Review by jufajardini — COMMENTED** [review]
Two very nit inline comments. 

About documentation, should we include anything in the upgrading section (here? https://docs.suricata.io/en/latest/upgrade.html#upgrading-7-0-to-8-0) about not using luajit anymore, or isn't that necessary?

File: .github/workflows/commits.yml [hunk 1]
```diff
@@ -85,7 +85,7 @@ jobs:
               git checkout $rev
               echo "Building rev ${rev}" | tee -a build_log.txt
               ./autogen.sh >> build_log.txt 2>&1
-              CC="sccache gcc" ./configure --enable-unittests >> build_log.txt 2>&1
+              CC="sccache gcc" ./configure --enable-unittests  --disable-lua >> build_log.txt 2>&1
```
[the remark this answers is not in the document]
Comment (J0eJ0h) [reply .github/workflows/commits.yml:88]: Lol, no.   I need a new keyboard :(

[the remark this answers is not in the document]
Comment (jufajardini) [reply .github/workflows/commits.yml:88]: ;_;

**Comment by catenacyber** [comment]
Was this replaced by https://github.com/OISF/suricata/pull/10317 right ?

**Pull request closed without merging**